controllers/infinispan_controller.go

```diff
@@ -672,19 +680,20 @@ func (reconciler *InfinispanReconciler) Reconcile(ctx context.Context, ctrlReque
 	}
 
 	if infinispan.HasSites() {
-		crossSiteViewCondition, err := r.GetCrossSiteViewCondition(podList, infinispan.GetSiteLocationsName(), cluster)
+		crossSiteViewCondition, err := r.GetCrossSiteViewCondition(podList, infinispan.GetSiteLocationsName(), curl)
 		if err != nil {
 			return ctrl.Result{}, err
 		}
 		// ISPN-13116 If xsite view has been formed, then we must perform state-transfer to all sites if a SFS recovery has occurred
 		if crossSiteViewCondition.Status == metav1.ConditionTrue {
 			podName := podList.Items[0].Name
+			curl.SetPodname(podName)
```

Comment:
I don't like a lot this way of configuring the target pod for the ispnClient, looks a kind of 'configuring by side effect' to me...
but I don't have a better option, though.

Reply:
Yeah I don't like it either! I was trying to avoid the need todo `ispnClient.New()` every time a pod changed as that will eventually have some logic to determine the required client version, so would result in unnecessary rest calls. I'll have a think this afternoon and see if I can improve this.

Reply:
I've replaced it with a new helper method `InfinispanForPod` which clones an existing curl client and updates it's podname to the required value